can-stache-key.js

```diff
@@ -110,31 +119,34 @@ observeReader = {
 
 			checkForObservableAndNotify(options, state, getObserves, prev, i-1);
 
-
-
 			type = typeof cur;
+			parentHasKey = objHasKeyAtIndex(prev, reads, i - 1);
```

Comment:
can we avoid this check until the last property read? If the key is `"foo.bar.zed"` we shouldn't have to check `objHasKeyAtIndex` for every step.

Reply:
I moved this so that it is only called for the last item or the first item that is undefined.